pallets/committee/src/benchmarking.rs

```diff
@@ -0,0 +1,216 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use crate::*;
+use frame_benchmarking::benchmarks_instance;
+use frame_support::{dispatch::DispatchResult, traits::UnfilteredDispatchable, StorageValue};
+use polymesh_common_utilities::{
+    benchs::{User, UserBuilder},
+    MaybeBlock,
+};
+use sp_std::prelude::*;
+
+const COMMITTEE_MEMBERS_NUM: usize = 10;
+const COMMITTEE_MEMBERS_MAX: u32 = 100;
+const PROPOSAL_PADDING_LEN: usize = 10_000;
+const PROPOSALS_NUM: u8 = 100;
+
+fn make_proposals_and_vote<T, I>(users: &[User<T>]) -> DispatchResult
+where
+    I: Instance,
+    T: Trait<I>,
+{
+    for i in 0..PROPOSALS_NUM {
+        let index = Module::<T, I>::proposal_count();
+        let proposal: <T as Trait<I>>::Proposal =
+            frame_system::Call::<T>::remark(vec![i + 1; PROPOSAL_PADDING_LEN]).into();
+        identity::CurrentDid::put(users[0].did());
+        Module::<T, I>::vote_or_propose(users[0].origin.clone().into(), true, Box::new(proposal))?;
+        if users.len() > 1 {
+            let hash = *Module::<T, I>::proposals()
+                .last()
+                .ok_or("missing last proposal")?;
+            // cast max N-1 additional votes for proposal #N
+            for (j, user) in users.iter().skip(1).take(i as usize).enumerate() {
+                // Vote for the proposal if it's not finalised.
+                if Module::<T, I>::voting(&hash).is_some() {
+                    identity::CurrentDid::put(user.did());
+                    Module::<T, I>::vote(user.origin.clone().into(), hash, index, j % 2 == 0)?;
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+fn make_members_and_proposals<T, I>(m: u32) -> Result<Vec<User<T>>, DispatchError>
+where
+    I: Instance,
+    T: Trait<I>,
+{
+    let members: Vec<_> = (0..m)
+        .map(|i| {
+            UserBuilder::<T>::default()
+                .generate_did()
+                .seed(i)
+                .build("member")
+        })
+        .collect();
+    Members::<I>::put(members.iter().map(|m| m.did()).collect::<Vec<_>>());
+    make_proposals_and_vote::<T, I>(&members)?;
+    Ok(members)
+}
+
+benchmarks_instance! {
+    _ {}
+
+    set_vote_threshold {
+        let n = 1;
+        let d = 2;
+        let origin = T::CommitteeOrigin::successful_origin();
+        let call = Call::<T, I>::set_vote_threshold(n, d);
+    }: {
+        call.dispatch_bypass_filter(origin)?;
+    }
```

Comment:
Because custom origins require custom dispatch.
```
  error[E0308]: mismatched types
     --> /home/vladi/remote-builds/8028138931963931287/pallets/committee/src/benchmarking.rs:78:1
      |
  78  | / benchmarks_instance! {
  79  | |     _ {}
  80  | |
  81  | |     set_vote_threshold {
  ...   |
  216 | |     }
  217 | | }
      | |_^ expected associated type, found enum `core::result::Result`
      |
      = note: expected associated type `<T as frame_system::Trait>::Origin`
                            found enum `core::result::Result<frame_system::RawOrigin<<T as frame_system::Trait>::AccountId>, <T as frame_system::Trait>::Origin>`
      = help: consider constraining the associated type `<T as frame_system::Trait>::Origin` to `core::result::Result<frame_system::RawOrigin<<T as frame_system::Trait>::AccountId>, <T as frame_system::Trait>::Origin>`
      = note: for more information, visit https://doc.rust-lang.org/book/ch19-03-advanced-traits.html
      = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```